R/misc.R

```diff
@@ -663,19 +663,12 @@ check_training_set <- function(x, rec, fresh, call = rlang::caller_env()) {
     }
     x <- rec$template
   } else {
-    in_data <- vars %in% colnames(x)
-    if (!all(in_data)) {
-      cli::cli_abort(
-        "Not all variables in the recipe are present in the supplied training \\
-        set: {.and {.var {vars[!in_data]}}}.",
-        call = call
-      )
-    }
     if (!is_tibble(x)) {
-      x <- as_tibble(x[, vars, drop = FALSE])
-    } else {
-      x <- x[, vars]
+      x <- as_tibble(x)
     }
+    recipes_ptype_validate(rec, new_data = x, stage = "prep", call = call)
+
+    x <- x[, vars]
```

Comment:
This seems to work just fine, but I would have expected this'd be implemented by just replacing the previous 666-673 with the call to `recipes_ptype_available()` and leaving the conditional subset as-is. Any reason why we did it this way?

Reply:
because `recipes_ptype_validate()` expects `x` to be a data.frame. So it didn't work when matrices were passed to `x`.

I have plans to streamline this in https://github.com/tidymodels/recipes/issues/1333